File: Framework/Php/framework/framework/collections/smartkeyvaluepair.php
<?php

/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */

namespace Smartkernel\Framework\Collections {
    /*
     * 键值对
     */

    class SmartKeyValuePair {

        //键
        public $Key;
        //值
        public $Value;

        /*
         * 构造函数
         * 
         * $param object $key，键
         * $param object $value，值
         * 
         */

        function __construct($key, $value) {
            $this->Key = $key;
            $this->Value = $value;
        }

        /*
         * 转换为字符串
         */

        function __toString() {
            return $this->Key . "=" . $this->Value;
        }

    }

}

File: Framework/Java/framework/framework/smartkernel/framework/collections/SmartKeyValuePair.java
/*
 * 项目：Smartkernel.Framework
 * 作者：曹艳白
 * 邮箱：caoyanbai@139.com
 */
package smartkernel.framework.collections;

/**
 * 键值对
 *
 */
public class SmartKeyValuePair<K, V> {

	/**
	 * 构造函数
	 * 
	 * @param key，Key
	 * @param value，Value
	 */
	public SmartKeyValuePair(K key, V value) {
		this.key = key;
		this.value = value;
	}

	private K key;
	private V value;

	/**
	 * 获取Key
	 * 
	 * @return，Key
	 */
	public K getKey() {
		return key;
	}

	/**
	 * 设置Key
	 * 
	 * @param key
	 *            ，Key
	 */
	public void setKey(K key) {
		this.key = key;
	}

	/**
	 * 获取Value
	 * 
	 * @return，Value
	 */
	public V getValue() {
		return value;
	}

	/**
	 * 设置Value
	 * 
	 * @param value
	 *            ，Value
	 */
	public void setValue(V value) {
		this.value = value;
	}

	@SuppressWarnings("unchecked")
	@Override
	public boolean equals(Object arg0) {
		if (arg0 == null) {
			return false;
		}

		SmartKeyValuePair<K, V> other = (SmartKeyValuePair<K, V>) arg0;
		return this.key.equals(other.key) && this.value.equals(other.value);
	}

	@Override
	public int hashCode() {
		return (this.key.hashCode() + "," + this.value.hashCode()).hashCode();
	}

}
